server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -152,4 +137,60 @@ private String getVideoId(String response) {
     // Only returning the first for now, potential to change if I optimize search results.
     return searchResults.get(0);
   }
+
+  /**
+   * Parses the comment JSON string and retrieves the comments.
+   *
+   * @param response The JSON string to be parsed.
+   * @return An ArrayList with each comment
+   */
+  private ArrayList<String> retrieveComments(String response) {
+    ArrayList<String> commentList = new ArrayList<>();
+
+    // Traversing to the items array
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonArray itemsArray = jObject.getAsJsonArray("items");
+
+    for (JsonElement el : itemsArray) {
+      // Traversing each item in the array
+      String topComment =
+          el.getAsJsonObject()
+              .getAsJsonObject("snippet")
+              .getAsJsonObject("topLevelComment")
+              .getAsJsonObject("snippet")
+              .get("textOriginal")
+              .toString();
+
+      // Remove all non-ASCII characters
+      commentList.add(topComment.replaceAll("[^\\x00-\\x7F]", ""));
+    }
+
+    return commentList;
+  }
+
+  /**
+   * Condenses array of comments into one large string, formatting it along the way.
+   *
+   * @param comments The array to be condensed.
+   * @return A properly formatted Strin
+   */
+  private String convertToString(ArrayList<String> comments) {
+    StringBuilder res = new StringBuilder();
+
+    for (String string : comments) {
+      string = string.replace("\"", "");
+      // Make sure each comment is treated as its own sentence
+      // Not sure char datatype works with regex so used String
+      String lastCharacter = string.substring(string.length() - 1);
+      if (!lastCharacter.matches(".|!|\\?")) {
+        string += ". ";
+      } else {
+        string += " ";
+      }
+      res.append(string);
+    }
+
+    return res.toString();
+  }
```

Comment:
yes!!! IT's time this finally got coded (instead of using a jank website haha XD )

Reply:
Hahahaha